include/core/common/result.hpp

```diff
@@ -0,0 +1,457 @@
+#pragma once
+/**
+ * Copyright 2020 Raphael Baier, The Args Team
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files
```

Comment:
I am not fine with this

Reply:
what is going on here ?

Reply:
this is in the include folder, (aka should not be comitted)